### DISCUSSION TITLE
Is it possible to use egui for headless rendering? any examples?

### DISCUSSION THREAD

**Discussion by olivewong** [emilk/egui#5435] [Q&A]
hi, I'd like to use egui for text/2d overlay elements on top of a wgpu renderer. My current implementation uses winit but I also would like it to work headless (ie no winit, no window). I saw that the existing integrations depend on DisplayHandle so I am wondering if there are examples that don't depend on the window

(I realize that using egui may be overkill for just text/overlay but I knew the dependencies would play nice with re_renderer since it's used in rerun's labels so am just curious if it will work out here)

**Comment by lucasmerlin** [comment]
You can have a look at what we do in egui_kittest, there is a wgpu renderer that works headless
